model/cover_graph.py

```diff
@@ -1,26 +1,55 @@
 class CoverGraph(object):
-
-    def __init__(self, nodes, verteces):
-        self.nodes = nodes
-        self.verteces = verteces
+    def __init__(self, root_vertex=None, end_vertex=None, vertices=[], edges=[]):
+        self.root_vertex = root_vertex
+        self.end_vertex = end_vertex
+        self.vertices = vertices
+        self.edges = edges
 
     def to_execution_path_tree(self):
         pass
-        # return execution_path_tree
-
 
-class Node(object):
-
-    def __init__(self, label=''):
-        self.label = label
-        # Needed here ?
-        # self.operation = operation (dans assign, skip, while, if)
+    def renamed_edges(self, old_vertex, new_vertex):
+        new_edges = []
+        for edge in self.edges:
+            edge.root_vertex = (
+                new_vertex
+                if edge.root_vertex == old_vertex
+                else edge.root_vertex
+            )
+            edge.child_vertex = (
+                new_vertex
+                if edge.child_vertex == old_vertex
+                else edge.child_vertex
+            )
+            new_edges.append(edge)
+        return new_edges
```

Comment:
I do not really understand what this method does ... I understood it's needed when we have a sequence of instructions, like to merge the existing edges, but does that create new edges ? If so what for ?

Reply:
Ok I'll try to explain this with the example of the creation of the grpah for the sequence node. You have 2 different graphs with one input and one output for both of them. The idea of the new graph for the sequence node is to take the output vertex of the first graph and replace it by the input of the second one. To achieve this you want to replace the first output by the second input in the edges list of the first graph. This method returns this new edges list in order to not change the original edges list. This idea is to replace the old vertex by the new vertex in all edges.

Reply:
Ok got it, quite clever 👍

Reply:
Ok got it, quite clever 👍